### PR TITLE
Articles

### PR DESCRIPTION
**Pull request by wawer77**
Article - comment - user relation resolved. Admin - regular user differentiation. No session management nor navigation - will be added together with the layout (navbars, etc,). Didn't delete the non-default gems, as wasn't sure if it doesn't crash the app.

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class ArticlesController < ApplicationController
+
+  def index
+    @articles =Article.all
```
Comment (oskarlakner) [inline app/controllers/articles_controller.rb]: please use rubocop ;)

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  def edit
+    @article = Article.find(params[:id])
+  end
+
+  def create
+    @article = Article.new(article_params)
+    @article.save
+    redirect_to @article
+    #if function for saving or not to be added
```
Comment (oskarlakner) [inline app/controllers/articles_controller.rb]: why is it tba? :P

File: app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    redirect_to @article
+  end
+
+  def destroy
+    @article=Article.find(params[:id])
+    @article.destroy
+    redirect_to articles_path
+    end
+
+  #PRIVATE methods
```
Comment (oskarlakner) [inline app/controllers/articles_controller.rb]: no needed

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  def new
+    @comment = current_user.comments.build(comment_params)
+  end
+
+  def create
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.create(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      flash[:success] = "Comment added"
```
Comment (oskarlakner) [inline app/controllers/comments_controller.rb]: I'd be better to pass flash msg to redirect_to

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    end
+  end
+
+  def edit
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.find(params[:id])
+  end
+
+  def update
+    @article = Article.find(params[:article_id])
```
Comment (oskarlakner) [inline app/controllers/comments_controller.rb:32]: please extract to a private method

File: app/views/articles/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+<p><%= @article.text %></p>
+<% if current_user.admin? %>
+<%= link_to 'Edit', edit_article_path %>
+<%= link_to 'Delete', article_path,
+              method: :delete,
+              data: { confirm: 'Are you sure?' } %>
+<% end %>
+              <%= link_to 'Back to the list', articles_path %><br>
+<% #COMMENTS %>
+<h2>Comments</h2>
```
Comment (oskarlakner) [inline app/views/articles/show.html.erb]: extract to partial

File: app/views/articles/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  <p>
+    <%= link_to 'Delete', [comment.article, comment], method: :delete %>
+    <%= link_to 'Edit', edit_article_comment_path(comment.article, comment) %>
+  </p>
+  <% end %>
+<% end %>
+
+<% if user_signed_in? %>
+<% #ADD A COMMENT %>
+<h2>Add a comment:</h2>
```
Comment (oskarlakner) [inline app/views/articles/show.html.erb]: extract to partial

File: app/controllers/articles_controller.rb (hunk 3 above)
Comment (wawer77) [reply app/controllers/articles_controller.rb]: Did you mean the comment?

**Pull request closed without merging**